ui/src/adapters/api/credentials.ts

```diff
@@ -87,34 +118,105 @@ export const credentialStatusParser = getStrictParser<CredentialStatus>()(
 export async function getCredential({
   credentialID,
   env,
+  identifier,
   signal,
 }: {
   credentialID: string;
   env: Env;
+  identifier: Identifier;
   signal?: AbortSignal;
 }): Promise<Response<Credential>> {
   try {
-    const response = await axios({
+    const response: { data: CredentialDetail } = await axios({
```

Comment:
You can not do this. We always need to assume that the response we get from the API is broken. This is why we create zod parsers and use them to parse the data. To access the data from the credential you need to parse the response from the server before.